internal/graph/check_test.go

```diff
@@ -2998,3 +3000,338 @@ func TestIteratorReadStartingFromUser(t *testing.T) {
+				tuple.NewTupleKey("group:4", "member", "user:maria"),
+			},
+			dsError: nil,
+			objectIDs: []string{
+				"8", "9",
+			},
+			context: map[string]interface{}{},
+			expectedResolveCheckResponse: &ResolveCheckResponse{
+				Allowed: false,
+				ResolutionMetadata: &ResolveCheckResponseMetadata{
+					DatastoreQueryCount: 1,
+				},
+			},
+			expectedError: false,
+		},
+		{
+			name: "non_empty_iterator_match_cond_not_match",
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+
+				type user
+				type group
+					relations
+						define member: [user with condX]
+				type document
+					relations
+						define viewer: [group#member]
+
+				condition condX(x: int) {
+					x < 100
+				}
+`),
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKeyWithCondition("group:1", "viewer", "user:maria", "condX", nil),
+				tuple.NewTupleKeyWithCondition("group:2", "viewer", "user:maria", "condX", nil),
+				tuple.NewTupleKeyWithCondition("group:3", "viewer", "user:maria", "condX", nil),
+				tuple.NewTupleKeyWithCondition("group:4", "viewer", "user:maria", "condX", nil),
+			},
+			dsError: nil,
+			objectIDs: []string{
+				"2", "3",
+			},
+			context: map[string]interface{}{
+				"x": 200,
+			},
+			expectedResolveCheckResponse: &ResolveCheckResponse{
+				Allowed: false,
+				ResolutionMetadata: &ResolveCheckResponseMetadata{
+					DatastoreQueryCount: 1,
+				},
+			},
+			expectedError: false,
+		},
+		{
+			name: "non_empty_iterator_match_cond_match",
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+
+				type user
+				type group
+					relations
+						define member: [user with condX]
+				type document
+					relations
+						define viewer: [group#member]
+
+				condition condX(x: int) {
+					x < 100
+				}
+`),
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKeyWithCondition("group:1", "viewer", "user:maria", "condX", nil),
+				tuple.NewTupleKeyWithCondition("group:2", "viewer", "user:maria", "condX", nil),
+				tuple.NewTupleKeyWithCondition("group:3", "viewer", "user:maria", "condX", nil),
+				tuple.NewTupleKeyWithCondition("group:4", "viewer", "user:maria", "condX", nil),
+			},
+			dsError: nil,
+			objectIDs: []string{
+				"2", "3",
+			},
+			context: map[string]interface{}{
+				"x": 1,
+			},
+			expectedResolveCheckResponse: &ResolveCheckResponse{
+				Allowed: false,
+				ResolutionMetadata: &ResolveCheckResponseMetadata{
+					DatastoreQueryCount: 1,
+				},
+			},
+			expectedError: false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			objectIDs := storage.NewSortedSet()
+			for _, objectID := range tt.objectIDs {
+				objectIDs.Add(objectID)
+			}
+
+			associatedFunc := buildCheckAssociatedObjects(&ResolveCheckRequest{
+				StoreID:              ulid.Make().String(),
+				AuthorizationModelID: ulid.Make().String(),
+				TupleKey:             tuple.NewTupleKey("document:1", "viewer", "user:maria"),
```

Comment:
don't we need to test different inputs here too?

Reply:
Nothing in this top level function should behave differently according with various model / request.  These various combination should be in the inner function such as iteratorReadStartingFromUser / NewConditionsFilteredTupleKeyIterator / tupleIDInSortedSet.  These combo should be tested in the inner function (which is the whole point of breaking them into smaller piece so we don't have a large combo in the higher up function).  If there are specific thing we want to test, we can add them here.  I will add a TTU here for sanity for now.